### PR TITLE
Add proper output option for the forked Java compiler

### PR DESCRIPTION
**Pull request by tgodzik**
Previously, we only added the output directory for the local Java compiler, which caused compile artifacts to pop up in the source directories. Now, we properly add the output directory option.

Fixes https://github.com/scalacenter/bloop/issues/1691

I also removed 2 tests that were commented out for a long while.

### CODE REVIEW

File: backend/src/main/scala/bloop/CompilerCache.scala [hunk 1]
```diff
@@ -211,13 +211,19 @@ final class CompilerCache(
             val newClasspathValue = newInvalidatedEntry.toFile + File.pathSeparator + classpathValue
             options(classpathValueIndex) = newClasspathValue
 
+            output.getSingleOutputAsPath match {
+              case p if p.isPresent => java.nio.file.Files.createDirectories(p.get)
+              case _ =>
+            }
```
Comment (kpodsiad) [inline backend/src/main/scala/bloop/CompilerCache.scala:217]:
```suggestion
            output.getSingleOutputAsPath.ifPresent{p => 
              java.nio.file.Files.createDirectories(p)
              ()
            }
```
is it better than `p.isPresent` and `p.get` or not?

File: frontend/src/test/scala/bloop/bsp/BspMetalsClientSpec.scala [hunk 2]
```diff
@@ -300,7 +300,13 @@ class BspMetalsClientSpec(
 
   test("compile with semanticDB") {
     TestUtil.withinWorkspace { workspace =>
-      val `A` = TestProject(workspace, "A", dummyFooScalaAndBarJavaSources)
+
+      object JavacOptions {
+        // This will cause to use the forked javac compiler
```
Comment (kpodsiad) [inline frontend/src/test/scala/bloop/bsp/BspMetalsClientSpec.scala:305]: why it'll cause forking? Does every option cause forking?

File: backend/src/main/scala/bloop/CompilerCache.scala (hunk 1 above)
Comment (tgodzik) [reply backend/src/main/scala/bloop/CompilerCache.scala:217]: Both don't look great to be honest and this follows the style in Zinc/sbt

File: frontend/src/test/scala/bloop/bsp/BspMetalsClientSpec.scala (hunk 2 above)
Comment (tgodzik) [reply frontend/src/test/scala/bloop/bsp/BspMetalsClientSpec.scala:305]: Yes, every `-J` option causes forking.

Comment (kpodsiad) [reply frontend/src/test/scala/bloop/bsp/BspMetalsClientSpec.scala:305]: If you add ` every -J option causes forking.` as a comment it'll be awesome

File: backend/src/main/scala/bloop/CompilerCache.scala (hunk 1 above)
Comment (kpodsiad) [reply backend/src/main/scala/bloop/CompilerCache.scala:217]: I personally don't like pattern matching when the side effect is executed only for one branch, it kind of blurs it, `asScala` + `foreach` would be great.

**Review by kpodsiad — APPROVED** [review]

File: backend/src/main/scala/bloop/CompilerCache.scala (hunk 1 above)
Comment (tgodzik) [reply backend/src/main/scala/bloop/CompilerCache.scala]: changed to `ifPresent` here and in other place (LocalCompiler) also.

File: frontend/src/test/scala/bloop/bsp/BspMetalsClientSpec.scala (hunk 2 above)
Comment (tgodzik) [reply frontend/src/test/scala/bloop/bsp/BspMetalsClientSpec.scala]: Added!

**Pull request merged**